homeassistant/components/shelly/update.py

```diff
@@ -203,6 +204,18 @@ async def async_install(
         else:
             LOGGER.debug("Result of OTA update call: %s", result)
 
+    def version_is_newer(self, latest_version: str, installed_version: str) -> bool:
+        """Return True if available version is newer then installed version."""
+        return AwesomeVersion(
+            latest_version,
+            find_first_match=True,
+            ensure_strategy=[AwesomeVersionStrategy.SEMVER],
+        ) > AwesomeVersion(
+            installed_version,
+            find_first_match=True,
+            ensure_strategy=[AwesomeVersionStrategy.SEMVER],
+        )
+
```

Comment:
If I comment this method all the tests pass, can we add/modify a test that will fail without this custom comparison? If someone will remove it in the future we have no way to know why we even added it at first.
Can you update the method description to explain why we use this strategy?

Reply:
Thx for your feedback.
Reviewed tests and updated method descrition.